edenai_apis/tests/utils/test_upload_s3.py

```
@@ -27,3 +40,159 @@ def test_upload_to_s3():
+    @pytest.mark.asyncio
+    @patch(
+        "edenai_apis.utils.upload_s3.aupload_file_bytes_to_s3", new_callable=AsyncMock
+    )
+    @patch("edenai_apis.utils.upload_s3.aiofiles.open")
+    async def test_reads_file_and_delegates(self, mock_open, mock_upload_bytes):
+        mock_file = AsyncMock()
+        mock_file.read = AsyncMock(return_value=b"file content")
+        mock_open.return_value.__aenter__.return_value = mock_file
+        mock_upload_bytes.return_value = "https://uploaded-url"
+
+        result = await aupload_file_to_s3("/tmp/test.pdf", "test.pdf")
```

Ruff (0.15.10)

[error] Probable insecure usage of temporary file or directory: "/tmp/test.pdf"

(S108)